### PR TITLE
Support old images without /.singularity.d/actions/exec

### PR DESCRIPTION
**Pull request by DrDaveD**
**Description of the Pull Request (PR):**

This PR restores the old behavior of 2.x when /.singularity.d/actions/exec does not exist in an old image.  Depends on pr #1970 to work exactly the same so that should be merged first.

**This fixes or addresses the following GitHub issues:**

None

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [ ] I have tested this PR locally with a `make test`
- [x] This PR is against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

File: src/runtime/engines/singularity/process.go [hunk 1]
```diff
@@ -39,6 +40,28 @@ func (engine *EngineOperations) StartProcess(masterConn net.Conn) error {
 	args := engine.EngineConfig.OciConfig.Process.Args
 	env := engine.EngineConfig.OciConfig.Process.Env
 
+	if args[0] == "/.singularity.d/actions/exec" {
+		if _, err := os.Stat(args[0]); os.IsNotExist(err) {
+			// for backward compatibility with old images
+			if args[1][0] != '/' {
```
Comment (bauerm97) [inline src/runtime/engines/singularity/process.go]: I think this is checking for absolute path? Would rather use `filepath.IsAbs(args[1])` to check for maximum portability

File: src/runtime/engines/singularity/process.go [hunk 2]
```diff
@@ -39,6 +40,28 @@ func (engine *EngineOperations) StartProcess(masterConn net.Conn) error {
...
+				// match old behavior of searching path
+				oldpath := os.Getenv("PATH")
+				for _, keyval:= range env {
```
Comment (bauerm97) [inline src/runtime/engines/singularity/process.go]: Make sure you're running `gofmt` over your code, this shouldn't be `keyval:= ...` but rather `keyval := ...`.

**Review by bauerm97 — CHANGES_REQUESTED** [review]
Just a few really minor nitpicks here @DrDaveD

File: src/runtime/engines/singularity/process.go (hunk 1 above)
Comment (DrDaveD) [reply src/runtime/engines/singularity/process.go]: Ok I changed it to that, thanks.

File: src/runtime/engines/singularity/process.go (hunk 2 above)
Comment (DrDaveD) [reply src/runtime/engines/singularity/process.go]: Ah, I wasn't aware of go fmt, very useful.  I think travis should run it so updates do not get accepted unless it has been run.

**Comment by DrDaveD** [comment]
I made requested changes and rebased on the current master.

**Review by bauerm97 — APPROVED** [review]
Looks good after requested changes were addressed

**Comment by GodloveD** [comment]
Thanks @DrDaveD !

**Pull request merged**